Dockerfile

```diff
@@ -1,8 +1,54 @@
 FROM tiangolo/uwsgi-nginx-flask:flask
 
-COPY * /app
+RUN apt-get update && apt-get upgrade --fix-missing -y
+RUN apt-get update && apt-get install -y curl git bzr mercurial build-essential
+RUN apt-get install -y zip ruby-full haskell-platform shellcheck ssh
+RUN apt-get install -y python-pip python-dev
+RUN apt-get install -y nodejs build-essential golang
 
-RUN curl -sL https://deb.nodesource.com/setup_6.x | sudo -E bash -
-RUN apt-get update && apt-get install -y ruby-full haskell-platform shellcheck nodejs build-essential nodejs-legacy
+ENV NVM_DIR /usr/local/nvm
+ENV NODE_VERSION node
 
+# Install nvm with node and npm
+RUN curl https://raw.githubusercontent.com/creationix/nvm/v0.33.0/install.sh | bash \
+    && . $NVM_DIR/nvm.sh \
+    && nvm install $NODE_VERSION \
+    && nvm alias default $NODE_VERSION \
+    && nvm use default \
+    && npm install -g jsonlint jscs eslint jshint
+
+ENV NODE_PATH $NVM_DIR/versions/node/v$NODE_VERSION/lib/node_modules
+ENV PATH      $NVM_DIR/versions/node/v$NODE_VERSION/bin:$PATH
+
+RUN echo '. "$NVM_DIR/nvm.sh"' >> /etc/profile
+RUN echo NODE_VERSION=$NODE_VERSION >> /etc/environment
+RUN echo NVM_DIR=$NVM_DIR >> /etc/environment
+RUN echo NODE_PATH=$NVM_DIR/versions/node/v$NODE_VERSION/lib/node_modules >> /etc/environment
+RUN echo PATH=$NVM_DIR/versions/node/v$NODE_VERSION/bin:$PATH >> /etc/environment
+
+RUN pip install -U pip
+RUN curl -L https://raw.githubusercontent.com/yyuu/pyenv-installer/master/bin/pyenv-installer | bash
+RUN export PATH="/root/.pyenv/bin:$PATH"
+ENV PATH /root/.pyenv/bin:$PATH
+RUN eval "$(pyenv init -)"
+RUN eval "$(pyenv virtualenv-init -)"
+RUN pyenv update
```

Comment:
You probably want the pyenv steps lower down in the dockerfile than the known hosts steps, since they're more likely to change. E.g. when you want to switch to a new python version.